File: Algorithm/Other/at coder/Other/A45/A45.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin", "utf8");

function main(input) {
	const [line1, line2] = input.trim().split("\n");
	const [N, target] = line1.split(" ");
	const colors = line2.trim().split("");

	// 2色を合成した結果を表すマップ
	const combine = {
		WW: "W",
		BB: "R",
		RR: "B",
		WB: "B",
		BW: "B",
		BR: "W",
		RB: "W",
		RW: "R",
		WR: "R",
	};

	// カードを左から順に1枚ずつ合成していく
	let current = colors[0];
	for (let i = 1; i < colors.length; i++) {
		const pair = current + colors[i];
		current = combine[pair];
	}

	console.log(current === target ? "Yes" : "No");
}

main(input);

File: Algorithm/Other/at coder/Other/atcoder/A45/A45.py
def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    N = int(data[0])
    target = data[1]
    colors = list(data[2])

    combine = {
        'WW': 'W',
        'BB': 'R',
        'RR': 'B',
        'WB': 'B',
        'BW': 'B',
        'BR': 'W',
        'RB': 'W',
        'RW': 'R',
        'WR': 'R',
    }

    current = colors[0]
    for i in range(1, N):
        pair = current + colors[i]
        current = combine[pair]

    print("Yes" if current == target else "No")

if __name__ == '__main__':
    main()
